training/src/main/scala/com/airbnb/aerosolve/training/TrainingUtils.scala

```diff
@@ -226,16 +227,33 @@ object TrainingUtils {
   }
 
   // Returns the statistics of a feature
-  case class FeatureStatistics(count: Double, min: Double, max: Double, mean: Double, variance: Double)
+  case class FeatureStatistics(
+                                count: Double,
+                                min: Double,
+                                max: Double,
+                                mean: Double,
+                                variance: Double,
+                                quantiles: Seq[Double]
+                              )
 
   /**
     * Compute [[FeatureStatistics]] across examples and ignore those with less than minCount features.
     */
-  def getFeatureStatistics(minCount: Int, input: RDD[Example]): Array[((String, String), FeatureStatistics)] = {
-    val sqlContext = SQLContext.getOrCreate(input.sparkContext)
+  def getFeatureStatistics(minCount: Int,
+                           input: RDD[Example],
+                           quantiles: Seq[Double] = Nil): Array[((String, String), FeatureStatistics)] = {
```

Comment:
This was checked in additive trainer instead because it makes sense to gather more quantiles out of the context of additive trainer. When we have `approxPercentile` in Spark 2.0, we will probably be able to afford to get a lot more quantiles and do a more efficient dynamic bucketing.